packages/openapi-fetch/test/types.test.ts

```diff
@@ -280,4 +286,179 @@ describe("types", () => {
       assertType<Response>({ error: "default application/json" });
     });
   });
+
+  describe("OpenApiStatusToHttpStatus", () => {
+    test("returns numeric status code", () => {
+      assertType<OpenApiStatusToHttpStatus<200, number>>(200);
+      assertType<OpenApiStatusToHttpStatus<200, string>>(200);
+      assertType<OpenApiStatusToHttpStatus<204, string>>(204);
+
+      assertType<OpenApiStatusToHttpStatus<204, string>>(
+        // @ts-expect-error 200 is not a valid
+        200,
+      );
+      assertType<OpenApiStatusToHttpStatus<204, number>>(
+        // @ts-expect-error 200 is not a valid
+        200,
+      );
+
+      assertType<OpenApiStatusToHttpStatus<404, 200 | 204 | 206 | 404 | 500 | "default">>(404);
+    });
+
+    test("returns default response", () => {
+      type Status = OpenApiStatusToHttpStatus<"default", 200 | 204 | 206 | 404 | 500 | "default">;
+      assertType<Status>(
+        // @ts-expect-error 200 has been manually defined
+        200,
+      );
+      assertType<Status>(
+        // @ts-expect-error 204 has been manually defined
+        204,
+      );
+      assertType<Status>(201);
+      assertType<Status>(504);
+    });
+
+    test("returns 200 likes response", () => {
+      type Status = OpenApiStatusToHttpStatus<"2XX", 200 | 204 | 206 | 404 | 500 | "default">;
```

Comment:
It feels to me like this test case is "degenerate" at least if I understand the usage in this PR correctly:

I would expect the `Status` parameter to always be a subtype of `AllStatuses` (which in this case it isn't).

For this test case specifically, I would expect that the explicitly / numerically defined status codes are *not* part of the end result (precedence, as I pointed out in the doc comment of `OpenApiStatusToHttpStatus`).